bpt_main.cpp

```diff
@@ -0,0 +1,66 @@
+#include <fstream>
+#include <iostream>
+#include <pin.H>
+#include "bpt.hpp"
+#include "bpt_inst_counter.hpp"
+#include "bpt_writer_text.hpp"
+#include "bpt_writer_frames.hpp"
+
+KNOB<string> tracefile(KNOB_MODE_WRITEONCE, "pintool",
+                       "o", "trace.frames",
+                       "Trace file to output to "
+                       "(filename.{frames | txt | cnt})");
+
+INT32 usage() {
+    PIN_ERROR( "This Pintool trace "
+               "instructions memory and registers usage\n"
+              + KNOB_BASE::StringKnobSummary() + "\n");
+    return -1;
+}
+
+VOID fini(INT32 code, VOID* ptr) {
+    bpt::visitor* out = static_cast<bpt::visitor*>(ptr);
+    bpt::fini(code, out);
+    delete out;
+}
+
+VOID trace(TRACE trace, VOID* ptr) {
+    bpt::visitor* out = static_cast<bpt::visitor*>(ptr);
+    bpt::trace(trace, out);
+}
+
+VOID modload(IMG img, VOID*) {
+    bpt::modload(img);
+}
+
+VOID syscall_entry(THREADID tid, CONTEXT* ctx,
+                   SYSCALL_STANDARD std, VOID*) {
+    bpt::syscall_entry(tid, ctx, std);
+}
+
+extern char** environ;
+
+int main(int argc, char *argv[]) {
+    PIN_InitSymbols();
+    if (PIN_Init(argc, argv)) return usage();
+    bpt::visitor* out;    
```

Comment:
well, although it is not a real bug, but it still kind of ugly to use raw pointer, and do not free it. Can you use some smart pointer here?